pkg/vdri/key/creator.go

```diff
@@ -61,7 +61,7 @@ func createDoc(pubKeyValue []byte) (*did.Doc, error) {
 		AssertionMethod:      []did.VerificationMethod{{PublicKey: *pubKey}},
 		CapabilityDelegation: []did.VerificationMethod{{PublicKey: *pubKey}},
 		CapabilityInvocation: []did.VerificationMethod{{PublicKey: *pubKey}},
-		KeyAgreement:         []did.VerificationMethod{{PublicKey: *keyAgreement}},
+		KeyAgreement:         []did.VerificationMethod{{PublicKey: *keyAgreement, Embedded: true}},
```

Comment:
I think we should also fill `Relationship` field here for consistency.
For the better user experience, I would add the following factory methods of the `VerificationMethod`:
```
func NewEmbeddedVerificationMethod(pk PublicKey, r VerificationRelationship) *VerificationMethod {
	return &VerificationMethod{
		PublicKey:    pk,
		Relationship: r,
		Embedded:     true,
	}
}

func NewReferencedVerificationMethod(pk PublicKey, r VerificationRelationship, isRelativeURL bool) *VerificationMethod {
	return &VerificationMethod{
		PublicKey:    pk,
		Relationship: r,
		RelativeURL:  isRelativeURL,
	}
}
```

and will use it here:
```
	return &did.Doc{
		Context:              []string{schemaV1},
		ID:                   didKey,
		PublicKey:            []did.PublicKey{*pubKey},
		Authentication:       []did.VerificationMethod{
			*did.NewReferencedVerificationMethod(*pubKey, did.Authentication, false)},
		AssertionMethod:      []did.VerificationMethod{
			*did.NewReferencedVerificationMethod(*pubKey, did.AssertionMethod, false)},
		CapabilityDelegation: []did.VerificationMethod{
			*did.NewReferencedVerificationMethod(*pubKey, did.CapabilityDelegation, false)},
		CapabilityInvocation: []did.VerificationMethod{
			*did.NewReferencedVerificationMethod(*pubKey, did.CapabilityInvocation, false)},
		KeyAgreement:         []did.VerificationMethod{
			*did.NewEmbeddedVerificationMethod(*keyAgreement, did.KeyAgreement)},
		Created:              &t,
		Updated:              &t,
	}, nil
```